File: src/main/java/net/arvandor/numinoustreasury/listener/InventoryClickListener.java
package net.arvandor.numinoustreasury.listener;

import net.arvandor.numinoustreasury.workstation.WorkstationInterface;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.inventory.Inventory;

public final class InventoryClickListener implements Listener {

    @EventHandler
    public void onInventoryClick(InventoryClickEvent event) {
        if (!(event.getView().getTopInventory().getHolder() instanceof WorkstationInterface workstationInterface)) return;
        event.setCancelled(true);
        Inventory clickedInventory = event.getClickedInventory();
        if (clickedInventory != null && clickedInventory.getHolder() instanceof WorkstationInterface) {
            workstationInterface.onClick(event.getSlot());
        }
    }

}

File: src/main/kotlin/net/arvandor/numinoustreasury/listener/InventoryClickListener.kt
package net.arvandor.numinoustreasury.listener

import net.arvandor.numinoustreasury.workstation.WorkstationInterface
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryClickEvent

class InventoryClickListener : Listener {
    @EventHandler
    fun onInventoryClick(event: InventoryClickEvent) {
        if (event.view.topInventory.holder !is WorkstationInterface) return
        event.isCancelled = true
        val clickedInventory = event.clickedInventory
        val inventoryHolder = clickedInventory?.holder
        if (inventoryHolder is WorkstationInterface) {
            inventoryHolder.onClick(event.slot)
        }
    }
}
